candle-transformers/src/models/whisper/audio.rs

```diff
@@ -180,10 +212,55 @@ fn log_mel_spectrogram_<T: Float + std::fmt::Display>(
         samples_padded
     };
 
-    // Use a single thread for now.
-    let mut mel = log_mel_spectrogram_w(
-        0, &hann, &samples, filters, fft_size, fft_step, speed_up, n_len, n_mel, 1,
-    );
+    // ensure that the number of threads is even and less than 12
+    let n_threads = std::cmp::min(get_num_threads() - get_num_threads() % 2, 12);
+
+    let hann = Arc::new(hann);
+    let samples = Arc::new(samples);
+    let cloned_contents_of_filters = filters.to_vec();
```

Comment:
What is the rationale for cloning the filters here? This seems a bit wasteful compare to just using the reference if we can.

Reply:
no good rational 😅 haha, I was originally having lifetime issues when passing filters `&[T]` between threads and forced it to work with the clone, I meant to remove before pushing but missed it. Thanks for pointing out 🙏

This should be resolved in the latest commit and relies on passing a reference into scoped threads. Please let me know if this should be updated!